src/lib/TranscriptEditor/index.js

```diff
@@ -98,26 +134,59 @@ class TranscriptEditor extends React.Component {
 
   handleShowTimecodes = (e) => {
     const isChecked = e.target.checked;
-
     this.setState({ showTimecodes: isChecked });
+
+    if (this.props.handleAnalyticsEvents !== undefined) {
+      this.props.handleAnalyticsEvents({ 
+        category: 'TranscriptEditor', 
+        action: 'handleShowTimecodes', 
+        name: 'showTimecodes', 
+        value: isChecked
+      });
+    }
   }
 
   handleShowSpeakers = (e) => {
     const isChecked = e.target.checked;
-
     this.setState({ showSpeakers: isChecked });
+
+    if (this.props.handleAnalyticsEvents !== undefined) {
+      this.props.handleAnalyticsEvents({ 
+        category: 'TranscriptEditor', 
+        action: 'handleShowSpeakers', 
+        name: 'showSpeakers', 
+        value:  isChecked
+      });
+    }
   }
 
   handleSettingsToggle = () => {
     this.setState(prevState => ({
       showSettings: !prevState.showSettings
     }));
+    if (this.props.handleAnalyticsEvents !== undefined) {
```

Comment:
keep it consistent - new line after the setState (like you've done on 181 👍)

Reply:
cheers